external/fv3fit/fv3fit/reservoir/reservoir.py

```diff
@@ -140,6 +142,10 @@ def dump(self, path: str) -> None:
         with fs.open(os.path.join(path, self._RESERVOIR_WEIGHTS_NAME), "wb") as f:
             scipy.sparse.save_npz(f, self.W_res)
 
+        if self.state is not None:
```

Comment:
Could you move the state dumping into a separate `dump_state` method that would be called separately from the rest of the model dump? That way, the regular training would still just save a model without a warm start.